user-interface/src/lib/hooks/UseState.ts

```diff
@@ -0,0 +1,14 @@
+import React, { Dispatch, SetStateAction } from 'react';
+
+// <S>(initialState: S | (() => S)): [S, Dispatch<SetStateAction<S>>];
+// <S = undefined>(initialState: S): [S | undefined, Dispatch<SetStateAction<S | undefined>>];
```

Comment:
Is this for documentation purposes or ... ?

Reply:
Not really. These are function type definitions from React.